File: webapp/src/types/document.ts
import type { Attendee } from './event'

export interface Document {
  id: string
  filename: string
  contentType: string
  sizeBytes: number
  familyMemberId: string | null
  uploadedAt: string
}

export interface ExtractedEventCandidate {
  id: string
  title: string
  startDt: string
  endDt: string
  allDay: boolean
  attendees: Attendee[]
}

const EXTRACTABLE_CONTENT_TYPES = new Set([
  'application/pdf',
  'image/jpeg',
  'image/png',
  'image/gif',
  'image/webp',
])

export function isExtractable(contentType: string): boolean {
  return EXTRACTABLE_CONTENT_TYPES.has(contentType)
}

File: backend/app/schemas/document.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field, field_validator

from app.schemas.event import AttendeeSchema


class DocumentUpdate(BaseModel):
    family_member_id: Optional[str] = None
    filename: Optional[str] = Field(None, min_length=1, max_length=255)

    @field_validator("filename")
    @classmethod
    def _strip_and_reject_blank_filename(cls, value: Optional[str]) -> Optional[str]:
        if value is None:
            return value
        trimmed = value.strip()
        if not trimmed:
            raise ValueError("filename must not be blank")
        return trimmed


class DocumentResponse(BaseModel):
    id: str
    filename: str
    content_type: str
    size_bytes: int
    family_member_id: Optional[str]
    uploaded_at: datetime

    model_config = {"from_attributes": True}


class ExtractedEvent(BaseModel):
    title: str
    start_dt: datetime
    end_dt: datetime
    all_day: bool = False
    attendees: list[AttendeeSchema] = Field(default_factory=list)


class ExtractEventsResponse(BaseModel):
    events: list[ExtractedEvent]
